### DISCUSSION TITLE
How to get proper indentation when pressing enter after autoclosing html tag for nvim-ts-autotag plugin

### DISCUSSION THREAD

**Discussion by 231tr0n** [LazyVim/LazyVim#1832] [Q&A] [answered]
Currently lazyvim is closing tags properly but the issue is that if I hit enter after tag completition it becomes something like this:-
```
<html>|</html>
```
Hit enter then it becomes
```
<html>
|</html>
```
How do I make it like this
```
<html>
  |
</html>
```

Could someone please help me in figuring out how to achieve this. Any help is really appreciated.

**Comment by dpetka2001** [comment, marked as answer]
Try this config
```lua
  {
    "hrsh7th/nvim-cmp",
    dependencies = {
      { "windwp/nvim-autopairs", opts = {} },
    },
    opts = function()
      local cmp = require("cmp")
      local cmp_autopairs = require("nvim-autopairs.completion.cmp")

      -- Insert parentheses after selecting method/function
      cmp.event:on("confirm_done", cmp_autopairs.on_confirm_done())
    end,
  },
```
For some reason `nvim-cmp` on its own seems to not have correct indentation for html tags, but with the help of `nvim-autopairs` you are able to remedy that.

**Reply by 231tr0n** [reply]
It works. Once again thank you. I just had one doubt like this seems like an important functionality of nvim-ts-autotag missing in lazyvim right why not add this by default.

**Reply by dpetka2001** [reply]
Because this is an issue of `nvim-cmp` I would guess? You'd be better off opening an issue at the plugin's repo. They will give you a more appropriate answer.

**Reply by 231tr0n** [reply]
Yup

**Reply by dpetka2001** [reply]
More specifically, it seems to be an issue with how `nvim-cmp` and `nvim-ts-autotag` interact with each other. Not sure which repo would be more appropriate to open an issue to be honest.

**Reply by 231tr0n** [reply]
Ohh you meant in the plugin repo ok my bad will do it there.

**Reply by moliyu** [reply]
@dpetka2001 How does this work in blink.cmp?
